sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -353,6 +367,30 @@ def create_or_update(
         name, version = component._get_rest_name_version()
         rest_component_resource = component._to_rest_object()
         result = None
+        try:
+            if not component._is_anonymous:
+                client_component_hash = component.get_component_hash(keys_to_omit=["version"])
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
```

Comment:
I'm afraid you'll need to update recordings for all e2etests involving named component registration as the hash will be added to request body.

Reply:
ok